File: Login App/MobApplication/app/src/main/java/com/example/mobapplication/MainActivity.java
package com.example.mobapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.Toast;

import es.dmoral.toasty.Toasty;

public class MainActivity extends AppCompatActivity {
    EditText email,password,name;
    CheckBox verify, rememberMe;
    //public static String EMAIL_VALUE = "";
    SharedPreferences preferences;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //setContentView(R.layout.activity_main); Can use to get hello world activity
        setContentView(R.layout.acitvity_mn);

        name = findViewById(R.id.name);
        email = findViewById(R.id.email);
        password = findViewById(R.id.password);
        verify = findViewById(R.id.verify);
        rememberMe = findViewById(R.id.remember_me);

        preferences = getSharedPreferences("Auth",MODE_PRIVATE);

        if(preferences.getBoolean("remember",true)){
            Intent intent = new Intent(this,WelcomeActivity.class);
            String email = preferences.getString("email", "DEFAULT");
            intent.putExtra("email",email);
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
            startActivity(intent);
        }
    }
    public void onLOGIN(View view){
        String nameEntered = name.getText().toString();
        String emailEntered = email.getText().toString();
        String passwordEntered = password.getText().toString();
        //EMAIL_VALUE = emailEntered;

        if(TextUtils.isEmpty(nameEntered) || TextUtils.isEmpty(emailEntered) || TextUtils.isEmpty(passwordEntered)){
            Toasty.error(this,getString(R.string.complete_all_fields),Toast.LENGTH_SHORT).show();
            
        }else{
            if(verify.isChecked()) {
                Intent intent = new Intent(this, WelcomeActivity.class);
                intent.putExtra("email", emailEntered);



                Toasty.success(this,getString(R.string.entered_details, emailEntered, passwordEntered), Toast.LENGTH_SHORT).show();
                // Shared preferences
                SharedPreferences.Editor editor = preferences.edit();
                editor.putString("name",nameEntered);
                editor.putString("email",emailEntered);
                editor.putString("password",passwordEntered);


                editor.putBoolean("remember", rememberMe.isChecked());
                editor.apply();
                //intent.putExtra("age", 80);
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                startActivity(intent);
            }
            else{
                Toasty.warning(this,getString(R.string.verify_checkbox), Toast.LENGTH_SHORT).show();
            }
        }
    }
}

File: Dice Roller App/HelloKotlin/app/src/main/java/com/example/hellokotlin/MainActivity.kt
package com.example.hellokotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    private lateinit var diceImage : ImageView

    

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val rollButton = findViewById<Button>(R.id.bt_roll)
        rollButton.setOnClickListener { rollDice() }


        diceImage = findViewById(R.id.iv_dice)

        

    }

    private fun rollDice(){

        

        diceImage.setImageResource(getRandomDiceImage())

        

        Toast.makeText(this, getString(R.string.dice_rolled), Toast.LENGTH_SHORT).show()

       

    }

    private fun getRandomDiceImage() : Int {

        return when ((1..6).random()) {
            1 -> R.drawable.dice_1
            2 -> R.drawable.dice_2
            3 -> R.drawable.dice_3
            4 -> R.drawable.dice_4
            5 -> R.drawable.dice_5
            else -> R.drawable.dice_6
        }
    }




}
